docs/designs/025-implement-dry-run-health-checks.md

```diff
@@ -0,0 +1,533 @@
+	// Rest of the existing code remains the same, using nonDryRunEvents instead of healthEvents.Events
+}
+```
+
+File: `platform-connectors/pkg/connectors/kubernetes/metrics.go`
+
+```go
+var (
+	// ... existing metrics ...
+	
+	// NEW: Track dry run events skipped by platform connector
+	DryRunEventsSkipped = promauto.NewCounter(
+		prometheus.CounterOpts{
+			Name: "nvsentinel_platform_connector_dry_run_events_skipped_total",
+			Help: "Total number of dry run/audit events skipped (no node conditions/events created)",
+		},
+	)
+)
+```
+
+---
+
+### Step 4: Event Exporter
+External systems need to know which events are dry run for proper monitoring.
+
+File: `event-exporter/pkg/transformer/cloudevents.go`
+
+Update `ToCloudEvent()` function to add new field `dryRun` so that from the metrics we can get to know which events were published in dry run mode:
+
+```go
+func ToCloudEvent(event *pb.HealthEvent, metadata map[string]string) (*CloudEvent, error) {
+	// ... existing timestamp, entities code ...
+
+	healthEventData := map[string]any{
+		"version":            event.Version,
+		"agent":              event.Agent,
+		"componentClass":     event.ComponentClass,
+		"checkName":          event.CheckName,
+		"isFatal":            event.IsFatal,
+		"isHealthy":          event.IsHealthy,
+		"message":            event.Message,
+		"recommendedAction":  event.RecommendedAction.String(),
+		"errorCode":          errorCodes,
+		"entitiesImpacted":   entities,
+		"generatedTimestamp": timestamp,
+		"nodeName":           event.NodeName,
+		"dryRun":             event.DryRun,  // NEW: Include dry run flag
+	}
+
+	// ... rest of method
+}
+```
+
+File: `event-exporter/pkg/metrics/metrics.go`
+
+```go
+var (
+	// ... existing metrics ...
+	
+	// NEW: Track dry run events exported
+	DryRunEventsExported = promauto.NewCounter(
+		prometheus.CounterOpts{
+			Name: "nvsentinel_event_exporter_dry_run_events_exported_total",
+			Help: "Total number of dry run/audit events exported (for observability)",
+		},
+	)
+)
+```
+
+---
+
+### Step 5: Fault Quarantine Manager (FQM)
+
+Update this module to skip processing dry-run events so that:
+- Node is not cordoned
+- Annotations and labels are not applied
+- `nodeQuarantined` status is not set (remains null)
+
+**Why skipping `nodeQuarantined` update matters:** FRM and NDR use pipeline filters that match on `nodeQuarantined` status. If this field is not set, their pipelines will exclude the event, and they won't see it.
+
+File: `fault-quarantine/pkg/reconciler/reconciler.go`
+
+Add skip logic in `ProcessEvent()` method:
+
+```go
+func (r *Reconciler) ProcessEvent(
+	ctx context.Context,
+	event *model.HealthEventWithStatus,
+	ruleSetEvals []evaluator.RuleSetEvaluatorIface,
+	rulesetsConfig rulesetsConfig,
+) *model.Status {
+	if shouldHalt := r.checkCircuitBreakerAndHalt(ctx); shouldHalt {
+		return nil
+	}
+
+	// NEW: Skip dry run events and update the metric
+	if event.HealthEvent != nil && event.HealthEvent.DryRun {
```

Comment:
Can't we handle this in the mongodb pipeline? Do we need to add a check in code?